internal/component/kafka/consumer.go

```diff
@@ -35,28 +37,111 @@ type consumer struct {
 
 func (consumer *consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
 	b := consumer.k.backOffConfig.NewBackOffWithContext(session.Context())
+	var messages []*sarama.ConsumerMessage
+	messages = make([]*sarama.ConsumerMessage, 0)
+	isBulkSubscribe := consumer.k.checkBulkSubscribe(claim.Topic())
 	for message := range claim.Messages() {
+		if isBulkSubscribe {
+			if len(messages) < consumer.k.bulkSubscribeConfig.MaxBulkCount {
+				messages = append(messages, message)
+			} else {
+				break
+			}
+		} else {
+			if consumer.k.consumeRetryEnabled {
+				if err := retry.NotifyRecover(func() error {
+					return consumer.doCallback(session, message)
+				}, b, func(err error, d time.Duration) {
+					consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v. Retrying...", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				}, func() {
+					consumer.k.logger.Infof("Successfully processed Kafka message after it previously failed: %s/%d/%d [key=%s]", message.Topic, message.Partition, message.Offset, asBase64String(message.Key))
+				}); err != nil {
+					consumer.k.logger.Errorf("Too many failed attempts at processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				}
+			} else {
+				err := consumer.doCallback(session, message)
+				if err != nil {
+					consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				}
+			}
+		}
+	}
+	if isBulkSubscribe {
 		if consumer.k.consumeRetryEnabled {
 			if err := retry.NotifyRecover(func() error {
-				return consumer.doCallback(session, message)
+				return consumer.doBulkCallback(session, messages, claim.Topic())
 			}, b, func(err error, d time.Duration) {
-				consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v. Retrying...", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				consumer.k.logger.Errorf("Error processing Kafka bulk messages: %s. Error: %v. Retrying...", claim.Topic(), err)
```

Comment:
As per other issue, this should be a warning?

Reply:
Changed to Warnf, as it will retry.